src/cudadecoder/cuda-online-pipeline-dynamic-batcher.cc

```diff
@@ -144,9 +145,20 @@ void CudaOnlinePipelineDynamicBatcher::BatcherThreadLoop() {
             curr_batch_->corr_ids, curr_batch_->h_all_waveform,
             curr_batch_->n_samples_valid, curr_batch_->is_first_chunk,
             curr_batch_->is_last_chunk);
-        n_chunks_not_done_.fetch_sub(curr_batch_->Size(),
-                                     std::memory_order_release);
 
+        {
+          // Update counts
+          std::lock_guard<std::mutex> lk(next_batch_and_backlog_m_);
+          n_chunks_not_done_.fetch_sub(curr_batch_->Size(),
+                                       std::memory_order_release);
+          for (int i = 0; i < curr_batch_->corr_ids.size(); i++) {
```

Comment:
You will receive compiler warnings complaining about comparing signed integer `i` with
unsigned type `corr_ids.size()`.

Reply:
Done